modules/tinymce/src/core/main/ts/autocomplete/AutocompleteContext.ts

```diff
@@ -14,7 +14,7 @@ export interface AutocompleteContext {
 
 const stripTriggerChar = (text: string, triggerCh: string) => text.substring(triggerCh.length);
 
-const findChar = (text: string, index: number, ch: string): Optional<number> => {
+const findTrigger = (text: string, index: number, ch: string): Optional<number> => {
```

Comment:
Please rename `ch` here as this is no longer a single character.

- [ ] Rename to match that this isn't a single character anymore

Reply:
This function also isn't very performant anymore since it has to do a `indexOf` call on every character which could be slower on long strings (granted it should stop at the first whitespace char). This code also runs on every keypress so we need to try and make sure it's fairly quick. I think we'd be better refactoring this a little so that it's something like this instead:
```ts
const findTrigger = (text: string, index: number, trigger: string): Optional<number> => {
  // Identify the `char` in, and start the text from that point forward. If there is ever any whitespace, fail
  let i: number;
  const firstChar = trigger.charAt(0)

  for (i = index - 1; i >= 0; i--) {
    const char = text.charAt(i);
    if (isWhitespace(char)) {
      return Optional.none();
    }

    if (firstChar === char && Strings.contains(text, trigger, i, index)) {
      break;
    }
  }

  return Optional.some(i);
};
```
This way we only do the `indexOf` check if the first character in the trigger is found first.

- [ ] Consider improving the performance of this logic a little, like in the above

Reply:
Actually looking more the `Autocompleters` logic also needs `ch` renamed to reflect it's no longer a single character and the `triggerChars` property in `AutocompleterDatabase` also needs renaming... maybe to just `triggers`?

- [ ] Rename anything related to `ch` in the autocompleter logic